source/server/http/admin.cc

```diff
@@ -326,15 +332,18 @@ void AdminImpl::statsAsPrometheus(const std::list<Stats::CounterSharedPtr>& coun
                                   Buffer::Instance& response) {
   for (const auto& counter : counters) {
     const std::string tags = AdminImpl::formatTagsForPrometheus(counter->tags());
-    response.add(fmt::format("# TYPE {0} counter\n", counter->tagExtractedName()));
-    response.add(
-        fmt::format("{0}{{{1}}} {2}\n", counter->tagExtractedName(), tags, counter->value()));
+    std::string metric_name = counter->tagExtractedName();
```

Comment:
You could share this code and below as it is now more complicated. It could be a small function that takes a Metric and the name (gauge/counter) to put into the type string. I don't feel that strongly about this if you don't want to do this one.

Reply:
Ah interesting, I originally was passing the whole vector, and that didn't work out so well. But I can try passing one metric into the helper method.

Reply:
it seems like the Metric class doesn't have the `value` method at the top level. Do you think it is worth changing the abstract class to add that in (despite Histograms not having it)?

Reply:
Nah just skip it. I would still do the other item though.

Reply:
You should still use the sanitize function here and below.